docs/en/04_Changelogs/4.0.0.md

```diff
@@ -109,6 +242,107 @@ Core template locations have moved - if you're including or overriding these
+<!-- after -->
+private static $allowed_actions = [
+    'suggest'
+];
+```
+
+#### Get on board with PSR-4 autoloading
+
+While not critical to an upgrade, SilverStripe 4.0 has adopted the [PS-4 autoloading](http://www.php-fig.org/psr/psr-4/)
+standard for the core modules, so it's probably a good idea to be consistent. 
+
+You can implement this in your composer configuration like so:
+
+```js
+    ...
+    "autoload": {
+        "psr-4": {
+            "SilverStripe\\Blog\\": "src/"
+        }
+    },
+    ...
+```
+Now you just need to ensure that each class site in the correct folder location 
+(including case sensitivity) to match its namespace. For example,
+ `SilverStripe\Blog\Model\BlogPost` should live at `src/Model/BlogPost.php`.
+
+Note that you don’t have to use `src/` as the root folder name. You can continue 
+to use `code/` if you want to. SilverStripe has adopted the PSR-4 approach and 
+have also started to use `src/` as a default folder location instead of 
+`code/`. If you’re going to change, it's probably a good time to do it while you're
+upgrading.
+
+For examples, take a look at the file/folder structure and to the 
+`composer.json` configuration in either the `framework` or `cms` modules.
+
+Please note that there are changes to template structure which in some cases 
+require templates to be in a folder location that matches the namespace of the class 
+that it belongs to, e.g. `themes/mytheme/templates/MyVendor/Foobar/Model/MyModel.ss`.
+
+#### Get on board with PSR-# logging, too
+
+One of the great changes that comes with SilverStripe 4 is the introduction of 
+[PSR-3](http://www.php-fig.org/psr/psr-3/) compatible logger interfaces. This 
+means we can use thirdparty services like Monolog.
+
+Note that the old `SS_Log` class has been renamed to `SilverStripe\Logging\Log`, 
+but has also had most of its logic stripped out and the remainder is marked as 
+deprecated.
+
+```ss
+<!-- before -->
+SS_Log::log('My error message', SS_Log::ERR);
+<!-- after -->
+Injector::inst()->get('Logger')->error('My error message');
+```
+
+If you call that multiple times in one class, you might want to add a getter 
+method to centralize it a little - but you probably should've had that already 
+in this case!
+
+##### Unit tests and PSR-3 loggers
+
+If you have some code that you've tested originally using `SS_Log`, and now moved 
+to using a Monolog Logger then you may notice occasions where your unit tests pass 
+but output a bunch of warnings, debug information, etc. to `stderr`. This is because
+ Monolog will use `stderr` as the default Handler unless you specify one.
+
+This isn't so much a regression as simply that you're now seeing things happening 
+in your code that didn't used to happen. You have a couple of options here. You can either:
+
+* Address that there's possibly a logic problem in your unit tests
+* Push a conveniently named `NullHandler` to monolog to quiet it.
+
+```php
+$logger = Injector::inst()->get('Logger');
```

Comment:
'Logger' service is out of date; Please use `Psr\Log\LoggerInterface::class` instead